addon/store.js

```diff
@@ -131,13 +123,41 @@ var Store = Ember.Object.extend({
 
             var attr = params[0];
             var value = options[attr];
-            var func = function(m) {
+
+            return this._findWithFilterFunc(type, function filterFunction(m) {
                 return get(m, attr) === value;
-            };
-            return this._findWithFilterFunc(type, func);
+            });
         }
+
         return this._findByIdComputed(type, options);
     },
+    findOne(type) {
+        var store = this;
+
+        return Ember.ObjectProxy.extend({
```

Comment:
I'm thinking of a way to remove this ObjectProxy.  Would you be against this?  My initial idea is implement a private object that hangs off the model (like content) and `attr` will key off that to for its gets and sets.  The only thing this means is that the actual model object that was pushed to the store will hang off this new property.  But I don't see this being an issue, do you?

This would allow `findOne` to set the underlying property, think similar to what content is doing now, but return an actual instance of the model instead of this simulating proxy.__proto__ = model like it's doing today